File: matlab_package/brainflow/examples/Spo2.m
BoardShim.enable_dev_board_logger();
DataFilter.enable_dev_data_logger();
DataFilter.set_log_file('data.log');

params = BrainFlowInputParams();
board_shim = BoardShim(int32(BoardIds.SYNTHETIC_BOARD), params);
board_id = int32(BoardIds.SYNTHETIC_BOARD);
preset = int32(BrainFlowPresets.DEFAULT_PRESET);
board_descr = BoardShim.get_board_descr(board_id, preset);
sampling_rate = int32(board_descr.sampling_rate);
board_shim.prepare_session();
board_shim.start_stream(45000, '');
pause(200);
board_shim.stop_stream();
nfft = DataFilter.get_nearest_power_of_two(sampling_rate);
data = board_shim.get_board_data(board_shim.get_board_data_count(preset), preset);
board_shim.release_session();

ppg_channels = board_descr.ppg_channels;
ir_channel = ppg_channels(1);
red_channel = ppg_channels(2);
ir_data = data(ir_channel, :);
red_data = data(red_channel, :);
spo2 = DataFilter.get_oxygen_level(ir_data, red_data, sampling_rate, 0.0, -37.663, 114.91);
heart_rate = DataFilter.get_heart_rate(ir_data, red_data, sampling_rate, 1024);

File: julia_package/brainflow/test/spo2.jl
using BrainFlow

params = BrainFlowInputParams()
board_shim = BrainFlow.BoardShim(BrainFlow.SYNTHETIC_BOARD, params)
board_descr = BrainFlow.get_board_descr(BrainFlow.SYNTHETIC_BOARD)
sampling_rate = board_descr["sampling_rate"]

BrainFlow.prepare_session(board_shim)
BrainFlow.start_stream(board_shim)
sleep(5)
BrainFlow.stop_stream(board_shim)
data = BrainFlow.get_board_data(board_shim)
BrainFlow.release_session(board_shim)

ppg_channels = board_descr["ppg_channels"]
data_ir = data[ppg_channels[1], :]
data_red = data[ppg_channels[2], :]

spo2 = BrainFlow.get_oxygen_level(data_ir, data_red, sampling_rate)
println(spo2)
